server/pulp/server/db/model/__init__.py

```diff
@@ -399,45 +409,44 @@ def pre_save_signal(cls, sender, document, **kwargs):
         document.last_updated = dateutils.now_utc_timestamp()
 
         # If content was set on this unit, copy the content into place
-        if document._source_location and document._relative_path:
+        if document._source_location:
             server_storage_dir = config.config.get('server', 'storage_dir')
-            platform_storage_dir = os.path.join(server_storage_dir, 'content',
-                                                document.unit_type_id)
-            target_location = os.path.join(platform_storage_dir, document._relative_path)
+            platform_storage_location = os.path.join(server_storage_dir, 'units',
```

Comment:
I think this is what you mentioned could be a property. That would be good probably.